bql/semantic/expression.go

```diff
@@ -90,11 +90,18 @@ type evaluationNode struct {
 }
 
 // format cell to a trimmed comparable string
-func formatCell(c *table.Cell) string {
+func formatCell(c *table.Cell) (string, error) {
 	if c.L != nil {
-		return strings.TrimSpace(c.L.ToComparableString())
+		return strings.TrimSpace(c.L.ToComparableString()), nil
 	}
-	return strings.TrimSpace(c.String())
+	if c.S != nil {
```

Comment:
Although I suggested to do this, I wonder if we could really always consider a string cell to be comparable to a text literal.
I think we will have to add extensive tests to check that there is no obscure usage that will break with this.

Reply:
I have just added more tests, mainly having comparisons involving ID bindings in HAVING clauses followed by AFTER/BETWEEN/BEFORE clauses (as a way of scrutinizing this use case that may become commonly used in the future).